exercise3/app/src/main/java/com/example/codechallange3/MainActivity.java

```diff
@@ -0,0 +1,59 @@
+package com.example.codechallange3;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.util.Log;
+import android.util.TypedValue;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+public class MainActivity extends AppCompatActivity {
+    private Button button;
+    private EditText editText;
+    private String mes;
+    private TextView textView1, textView2;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        button = (Button) findViewById(R.id.activity1_button);
+        editText = (EditText) findViewById(R.id.activity1_editext);
+
+        if (getIntent().hasExtra("Message")) {
```

Comment:
For Message key use a constant and the text sizes should be defined in the dimen.xml file